packages/shared/src/components/modals/SlackIntegrationModal.tsx

```diff
@@ -0,0 +1,351 @@
+            type={TypographyType.Title1}
+            bold
+            color={TypographyColor.Primary}
+          >
+            Squad + Slack = 🔥
+          </Typography>
+          <Typography
+            type={TypographyType.Body}
+            color={TypographyColor.Tertiary}
+          >
+            Get instant updates in Slack and keep the conversation going!
+          </Typography>
+          <Button
+            className="w-full"
+            type="button"
+            variant={ButtonVariant.Primary}
+            size={ButtonSize.Large}
+            onClick={onConnectNew}
+          >
+            Connect to Slack
+          </Button>
+          <Button
+            className="w-full"
+            tag="a"
+            type="button"
+            variant={ButtonVariant.Float}
+            size={ButtonSize.Medium}
+            href={slackIntegration}
+            target="_blank"
+          >
+            Read more ➔
+          </Button>
+        </Modal.Body>
+      </Modal>
+    );
+  }
+
+  return (
+    <Modal {...modalProps}>
+      <ModalClose className="top-4" onClick={props.onRequestClose} />
+      <Modal.Body className="flex flex-col gap-4">
+        <div className="mr-auto flex items-center justify-center">
+          <SourceAvatar source={source} size={ProfileImageSize.Small} />
+          <Typography
+            type={TypographyType.Callout}
+            color={TypographyColor.Primary}
+          >
+            <Typography
+              tag={TypographyTag.Span}
+              bold
+              color={TypographyColor.Primary}
+            >
+              {source.name}
+            </Typography>{' '}
+            <Typography
+              tag={TypographyTag.Span}
+              color={TypographyColor.Tertiary}
+            >
+              @{source.handle}
+            </Typography>
+          </Typography>
+        </div>
+        <Typography
+          tag={TypographyTag.H3}
+          type={TypographyType.Title2}
+          color={TypographyColor.Primary}
+          bold
+        >
+          Configure Integration
+        </Typography>
+        {!!slackIntegrations?.length && (
+          <Dropdown
+            placeholder="Select workspace"
+            shouldIndicateSelected
+            buttonSize={ButtonSize.Medium}
+            iconOnly={false}
+            key="feed"
+            selectedIndex={selectedIntegrationIndex}
+            renderItem={(value, index) => {
+              const currentIntegration = slackIntegrations[index];
+              const ItemIcon =
+                currentIntegration.id === 'new' ? PlusIcon : SlackIcon;
+
+              return (
+                <span className="flex gap-2 typo-callout">
+                  <ItemIcon />
+                  {slackIntegrations[index].name}
+                </span>
+              );
+            }}
+            icon={<SlackIcon />}
+            options={slackIntegrations?.map((item) => item.name)}
+            onChange={(value, index) => {
```

Comment:
Quite a lot of big inline functions, can we somehow maybe move them top level or to a hook?